HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -156,6 +199,34 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
     reco::SuperClusterRef scRef = candidate.superCluster();
     if (scRef.isNull() && !scRef.isAvailable())
       continue;
+
+    reco::CaloClusterPtr SCseed = candidate.superCluster()->seed();
+    const EcalRecHitCollection* rechits = (std::abs(scRef->eta()) < 1.479) ? rechitsEB.product() : rechitsEE.product();
+    Cluster2ndMoments moments = EcalClusterTools::cluster2ndMoments(*SCseed, *rechits);
+    float sMin = moments.sMin;
+    float sMaj = moments.sMaj;
+
+    unsigned int seedId = (*SCseed).seed();
+
+    std::vector<DetId> mDetIds = EcalClusterTools::matrixDetId((topology), (*SCseed).seed(), rechitMatrixSize);
+
+    int size = pow(2 * rechitMatrixSize + 1, 2);
+    int detSize = mDetIds.size();
+    std::vector<float> mEnergies(size, 0.);
+    std::vector<float> mTimes(size, 0.);
+
+    for (int i = 0; i < detSize; i++) {
```

Comment:
what happens if `detSize > size` ?

Reply:
size is the maximum of the matrix size https://github.com/cms-sw/cmssw/blob/CMSSW_11_2_X/RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h

It turns out sometimes it is smaller. I am changing now that the size of the vector stored in scouting corresponds to the size of the vector returned by EcalClusterTools function